services/server/src/server/services/VerificationService.ts

```diff
@@ -345,13 +352,96 @@ export class VerificationService {
     return verificationId;
   }
 
+  public async verifyFromSimilarityViaWorker(
+    verificationEndpoint: string,
+    chainId: string,
+    address: string,
+  ): Promise<VerificationJobId> {
+    let runtimeBytecode: string;
+    try {
+      runtimeBytecode =
+        await this.sourcifyChainMap[chainId].getBytecode(address);
+    } catch (error) {
+      throw new VerificationError({
+        code: "cannot_fetch_bytecode",
+        chainId,
+        address,
+      });
+    }
+
+    const verificationId = await this.storageService.performServiceOperation(
+      "storeVerificationJob",
+      [new Date(), chainId, address, verificationEndpoint],
+    );
+
+    const sourcifyDatabaseService = this.storageService.rwServices[
+      "SourcifyDatabase"
+    ] as SourcifyDatabaseService;
+
+    sourcifyDatabaseService
+      .getSimilarityCandidatesByRuntimeCode(
+        runtimeBytecode,
+        DEFAULT_SIMILARITY_CANDIDATE_LIMIT,
+      )
+      .then(async (candidates) => {
+        if (candidates.length === 0) {
+          logger.info("No similarity candidates found", {
+            chainId,
+            address,
+          });
+          await this.storageService.performServiceOperation("setJobError", [
+            verificationId,
+            new Date(),
+            {
+              customCode: "no_similar_match_found",
+              errorId: uuidv4(),
+            },
+          ]);
+          return;
+        }
+
+        const creatorTxHash =
+          (await getCreatorTx(this.sourcifyChainMap[chainId], address)) ||
+          undefined;
+
+        const input: VerifySimilarityInput = {
+          chainId,
+          address,
+          runtimeBytecode,
+          creatorTxHash,
+          candidates,
+          traceId: asyncLocalStorage.getStore()?.traceId,
+        };
+
+        this.verifyViaWorker(verificationId, "verifySimilarity", input);
+      })
+      .catch(async (error) => {
```

Comment:
same here. Just return the promise instead of awaiting it such that it is properly added to the promise chain.